provider/azure/instancetype.go

```diff
@@ -525,7 +525,11 @@ func (env *azureEnviron) findInstanceSpec(
 	instanceTypesMap map[string]instances.InstanceType,
 	constraint *instances.InstanceConstraint,
 	imageStream string,
+	preferGen1Image bool,
 ) (*instances.InstanceSpec, error) {
+	if !constraint.Constraints.HasArch() && constraint.Arch != "" {
+		constraint.Constraints.Arch = &constraint.Arch
+	}
 
 	if constraint.Arch != arch.AMD64 {
```

Comment:
Do we still need this? I thought we were going to support arm?

Reply:
Not in this PR, maybe in future PRs.